### PR TITLE
Add s3 avatars storage support for Jira

### PR DESCRIPTION
**Pull request by asezavar**
## Pull request description

Jira S3 storage support for avatars

## Checklist
- [x] I have added unit tests
- [x] I have applied the change to all applicable products
- [x] The E2E test has passed (use `e2e` label)
- [x] (Atlassian only) Internal Bamboo CI is passing

### CODE REVIEW

**Review by bianchi2 — APPROVED** [review]

File: src/main/charts/jira/values.yaml [hunk 1]
```diff
@@ -533,6 +542,24 @@ jira:
     #
     localHomeSubPath: "log"
 
+  # Use AWS S3 to store attachments. From Confluence 8.1 onwards.
+  #
+
+  s3AvatarsStorage:
```
Comment (errcode1202) [inline src/main/charts/jira/values.yaml:548]: Just wondering what happens when attachments for s3 needs to be added? Do we create a new stanza like this i.e.

```
s3AttachmentStorage
  bucketName
  bucketRegion
  endpointOverride
```

Or is it better to just rename this stanza to `s3Storage` and its used when storing avatars or attachments or both?

File: src/main/charts/jira/values.yaml [hunk 2]
```diff
@@ -533,6 +542,24 @@ jira:
     #
     localHomeSubPath: "log"
 
+  # Use AWS S3 to store attachments. From Confluence 8.1 onwards.
```
Comment (errcode1202) [inline src/main/charts/jira/values.yaml:545]: Comment is Confluence specific

**Review by errcode1202 — CHANGES_REQUESTED** [review]

File: src/main/charts/jira/values.yaml (hunk 1 above)
Comment (bianchi2) [reply src/main/charts/jira/values.yaml:548]: I think it is possible that different buckets may be used for avatars and attachments. Maybe:

```
s3Storage:
  avatars:
    bucketName
    bucketRegion
    endpointOverride
  attachments:
    bucketName
    bucketRegion
    endpointOverride
```

File: src/test/java/test/JiraS3EnabledTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                product + ".s3AvatarsStorage.bucketName", "my-bucket",
+                product + ".s3AvatarsStorage.bucketRegion", "my-region"
+        ));
+
+        resources.getStatefulSet(product.getHelmReleaseName())
+                .getContainer()
+                .getEnv()
+                .assertHasValue("ATL_S3AVATARS_BUCKET_NAME", "my-bucket")
+                .assertHasValue("ATL_S3AVATARS_REGION", "my-region")
+                .assertDoesNotHaveAnyOf("ATL_S3AVATARS_ENDPOINT_OVERRIDE");
```
Comment (errcode1202) [inline src/test/java/test/JiraS3EnabledTest.java:68]: Nit: maybe this test can be removed (kinda a duplicate of `jira_s3_avatars_storage_env_vars`) and to cover the last assertion add `.assertDoesNotHaveAnyOf("ATL_S3AVATARS_ENDPOINT_OVERRIDE");` to `jira_s3_avatars_storage_env_vars`?

File: src/main/charts/jira/values.yaml (hunk 1 above)
Comment (errcode1202) [reply src/main/charts/jira/values.yaml:548]: Oh interesting. Is this an option @asezavar?

Comment (bianchi2) [reply src/main/charts/jira/values.yaml:548]: Of course, if we expect attachments to arrive soon :)

Comment (asezavar) [reply src/main/charts/jira/values.yaml:548]: @errcode1202 yes that's the plan

Comment (errcode1202) [reply src/main/charts/jira/values.yaml:548]: Should the stanza be updated now then to accomadate attachments?

Comment (asezavar) [reply src/main/charts/jira/values.yaml]: I just changed the s3 storage values structure based on your suggestion, but I haven't added the attachment part, and there will be another ticket for that.

**Review by bianchi2 — APPROVED** [review]

**Review by errcode1202 — APPROVED** [review]
🚀

**Pull request merged**